File: backend/app/schemas/note.py
from datetime import datetime

from pydantic import BaseModel


class NoteCreate(BaseModel):
    title: str
    content: str = ""


class NoteImport(BaseModel):
    title: str
    content: str


class NoteRename(BaseModel):
    title: str


class NoteContentUpdate(BaseModel):
    content: str


class NoteFlags(BaseModel):
    favorite: bool | None = None
    pinned: bool | None = None


class NoteMove(BaseModel):
    target_workspace_id: str


class Note(BaseModel):
    id: str
    title: str
    created_at: datetime
    updated_at: datetime
    favorite: bool = False
    pinned: bool = False


class NoteDetail(Note):
    content: str


class NoteListItem(Note):
    # A short content preview for the note list/search UI — centered on the
    # matched query when there's a content match, otherwise just the start
    # of the note. Empty for a brand-new empty note.
    snippet: str = ""

File: packages/types/src/note.ts
// ─── Notes ────────────────────────────────────────────────────────────────────
export interface NoteCreate {
  title: string
  content?: string
}

export interface NoteImport {
  title: string
  content: string
}

export interface NoteRename {
  title: string
}

export interface NoteContentUpdate {
  content: string
}

export interface NoteFlags {
  favorite?: boolean | null
  pinned?: boolean | null
}

export interface NoteMove {
  target_workspace_id: string
}

export interface NoteFolderUpdate {
  folder?: string | null
}

export interface NoteTagsUpdate {
  tags: string[]
}

export interface NoteCommentsUpdate {
  comments: NoteComment[]
}

export interface NoteComment {
  id: string
  text: string
  created_at: string
  resolved: boolean
}

export interface Note {
  id: string
  title: string
  created_at: string
  updated_at: string
  favorite: boolean
  pinned: boolean
  folder: string | null
  tags: string[]
  trashed: boolean
  trashed_at: string | null
  opened_at: string | null
  comments?: NoteComment[]
}

export interface NoteDetail extends Note {
  content: string
}

export interface NoteListItem extends Note {
  snippet: string
}
